### DISCUSSION TITLE
How do I get the function name when I create a new thread?

### DISCUSSION THREAD

**Discussion by hq1119418168** [micropython/micropython#16408] [General]
For example：
import _thread
def th_func1():
    print('thread th_func1 is running')
_thread.start_new_thread(th_func1, ())

In mod_thread_start_new_thread, the th_func1 is passed in as a function pointer：

    // set the function for thread entry
    th_args->fun = args[0];

how do I get the string "th_func1"?

**Comment by shariltumin** [comment]
I do not know an easy way to get the function name directly. You can use ```_thread.get_ident()``` to store the function name associated with a thread in a directory at ```_thread.start_new_thread()``` time. Consider the following:

```py
import _thread as th
import time

def fun1(n):
    ...
    print('ME:', th.get_ident(), tt[th.get_ident()])
    tt.pop(th.get_ident())

def fun2(c,n):
    ...
    print('ME:', th.get_ident(), tt[th.get_ident()])
    tt.pop(th.get_ident())

tt = {}

tt[th.start_new_thread(fun1, (500,))] = 'fun1'
tt[th.start_new_thread(fun2, ('A', 3))] = 'fun2'
tt[th.start_new_thread(fun2, ('B', 7))] = 'fun2'

print('Threads:', tt)

# main thread wait until no more tasks
while tt: time.sleep(10)
print('All done!')

```

**Reply by hq1119418168** [reply]
I'm actually trying to get the python-defined function name in the mod_thread_start_new_thread function in the modthread.c file
